### PR TITLE
init_posts(): load more posts on window resize

### PR DESCRIPTION
**Pull request by Frenzie**
Fixes:

1. Open FreshRSS in a shorter window
2. Resize to be much longer
3. Half the window remains empty

This is most obviously a problem on vertical monitors.

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested
- [ ] unit tests written (optional if too hard)
- [ ] documentation updated

[Additional information can be found in the documentation](https://github.com/FreshRSS/FreshRSS/tree/edge/docs/en/developers/04_Pull_requests.md).

### CODE REVIEW

**Comment by Alkarex** [comment]
Maybe not a problem because it is probably not something that happens so often, but it could be good to throttle the number of events, just like we do for onscroll:

image

File: p/scripts/main.js [hunk 1]
```diff
@@ -735,20 +735,23 @@ function onScroll() {
...
-		(box_to_follow === document.scrollingElement ? window : box_to_follow).onscroll = function () {
-			clearTimeout(timerId);
-			if (lastScroll + 500 < Date.now()) {
-				lastScroll = Date.now();
-				onScroll();
-			} else {
-				timerId = setTimeout(onScroll, 500);
-			}
-		};
+		(box_to_follow === document.scrollingElement ? window : box_to_follow).onscroll = debouncedOnscroll;
```
Comment (Frenzie) [inline p/scripts/main.js]:
```suggestion
		(box_to_follow === document.scrollingElement ? window : box_to_follow).onscroll = debouncedOnScroll;
```

**Comment by Frenzie** [comment]
I just noticed this also happens on first load.

Edit: different issue; it just doesn't load *enough*.

**Comment by Frenzie** [comment]
In the simplest terms something like this for the other problem, but that can probably be tackled better. In any case it's not quite part of this PR (unless you want it to be).
```diff
diff --git a/p/scripts/main.js b/p/scripts/main.js
index 4ddafb2b..d58303c3 100644
--- a/p/scripts/main.js
+++ b/p/scripts/main.js
@@ -1793,6 +1793,20 @@ function load_more_posts() {
                }
 
                load_more = false;
+
+               if (context.auto_load_more) {
+                       if (box_to_follow.offsetHeight > 0 &&
+                               box_to_follow.scrollTop + box_to_follow.offsetHeight + (window.innerHeight / 2) >= streamFooter.offsetTop) {
+                               // Too close to the last pre-loaded article
+                               load_more_posts();
+                       } else {
+                               const after = document.querySelectorAll('.flux.current ~ .flux').length;
+                               if (after > 0 && after <= 5) {
+                                       // Too few pre-loaded articles
+                                       load_more_posts();
+                               }
+                       }
+               }
        };
        req.send();
 }
```

**Comment by Alkarex** [comment]
Ok, let's make another PR for the other problem

**Pull request merged**